state/mongodb/mongodb.go

```diff
@@ -195,10 +206,28 @@ func (m *MongoDB) setInternal(ctx context.Context, req *state.SetRequest) error
 		filter[etag] = uuid.NewString()
 	}
 
-	update := bson.M{"$set": bson.M{id: req.Key, value: v, etag: uuid.NewString()}}
-	_, err := m.collection.UpdateOne(ctx, filter, update, options.Update().SetUpsert(true))
+	reqTTL, err := stateutils.ParseTTL(req.Metadata)
+	if err != nil {
+		return fmt.Errorf("failed to parse TTL: %w", err)
+	}
 
-	return err
+	var ttl any
+	if reqTTL != nil {
+		ttl = primitive.NewDateTimeFromTime(time.Now().Add(time.Second * time.Duration(*reqTTL)))
+	}
+
+	update := bson.D{{"$set", bson.D{
+		{id, req.Key},
+		{value, v},
+		{etag, uuid.NewString()},
```

Comment:
`uuid.NewString()` can panic. Please use `obj, err := uuid.NewRandom(); str := obj.String()` so you can check the error